### DISCUSSION TITLE
OpenBSD install instructions for UTM on arm64/M-series

### DISCUSSION THREAD

**Discussion by michealp-coder** [utmapp/UTM#5643] [General]
**Can a documentation page be created to instruct how to install OpenBSD on an M-series cpu.** 

I’ve read that others have been able to get UTM to boot OpenBSD on m-series but I can’t replicate. 

If the **install steps** could be documented, it’d be very helpful. 

————

Below are some links that might be useful:

https://www.openbsd.org/faq/faq4.html#Download

https://codeofconnor.com/running-an-arm64-openbsd-virtual-machine-on-apple-silicon-with-qemu/

**Comment by macuserguru** [comment]
A little bit Off Topic - search for optimally UTM settings for a FreeBSD aarch64 installation - and how to bring the mouse to work in GNOME desktop!
Basic installation of FreeBSD aarch64 works fine.

**Comment by michealp-coder** [comment]
Is there a particular article you recommend?

After doing that search and a few other searches, I'm not finding the
settings needed to boot OpenBSD with UTM on Apple Silicon.

**Reply by sysaulab** [reply]
In interest of helping newcomers experiencing problems with openbsd, I created a preinstalled OpenBSD VM repository at https://github.com/sysaulab/UTMBSD/releases/tag/openbsd

**Reply by MagnetarRocket** [reply]
Alas, still plagued by the viogpu problem.

**Comment by ideologysec** [comment]
2023-10-16 tested working with UTM 4.4.3 (beta) and OpenBSD 7.4, working.

Download OpenBSD 7.4 arm64 install img, then in UTM:

New -> Virtualize -> Other -> "Browse" to boot img ->
Set memory and CPU (I set 2048MB and 2 CPU cores) -> Set drive size (I chose 15GB) ->
Choose shared directory settings (I skipped) ->
Summary: check "Open VM Settings" ->
Navigate to Drives / "USB Drive" ->
Change Image Type from "CD/DVD (ISO)" to "Disk Image" -> _**(this is the critical piece)**_
Navigate to Network -> change Emulated Network card from "virtio-net-pci" to "virtio-net-device" -> _**(networking will fail without this)**_
Save, start the VM (make a clone first for testing purposes IMHO)

**Reply by jholland-vin-dit** [reply]
what about getting X11 to work?

**Reply by ideologysec** [reply]
I have not had any luck so far. OpenBSD does not appear to support VirtIO drivers for displays (even the newest version is only for consoles).

Reports on the internet seem to indicate that it supports the QEMU VNC driver, but UTM does not.

If you find out, feel free to share.

**Reply by gretel** [reply]
openbsd 7.3 has been working fine, 7.4 stopped. tried most `display` device setting to no success.

**Reply by gretel** [reply]
removing the `display` device in trade for a `serial` one set for the built-in terminal works around the issue for me
![Screenshot 2023-10-21 at 01 59 01](https://github.com/utmapp/UTM/assets/80815/4f685c38-c56a-4d83-b59b-e694891ee982)

**Comment by jholland-vin-dit** [comment]
￼





this setting works for me, but no X11, just text console.

**Reply by gretel** [reply]
X11 requires a graphical interface which a text console is not 😏

**Comment by michealp-coder** [comment]
I came across these instructions below and it helped.

https://news.ycombinator.com/item?id=37353853

For OpenBSD, the installer works after adjusting the devices in the guest.

https://cdn.openbsd.org/pub/OpenBSD/7.3/arm64/install73.img

  - UTM / Create New Virtual Machine / Virtualize / Other
  - Select 'Skip ISO boot' / Continue
  - Hardware / (adjust as needed) Continue
  - Storage / (adjust as needed) Continue
  - Shared Directory / Continue
  - Select 'Open VM Settings' / Save
  - Right-click on Devices/Display and select Remove
  - Devices / New / Serial
  - Drive / New / Import and select install73.img
  - Edit the VM again and move the second VirtIO Drive to the first
position by right-clicking on it and selecting 'Move Up'

**Reply by gretel** [reply]
op intends to use the `display` device, not remove it. for me this was working fine using 7.3 and stopped when upgrading to the now current 7.4.

**Reply by sysaulab** [reply]
For my freebsd guest in utm/m1 i use serial console to boot, xrdp server in freebsd guest accessed with Microsoft Remite Desktop app in macos over the network.

**Reply by benjamineskola** [reply]
> this was working fine using 7.3 and stopped when upgrading to the now current 7.4.

Just found the same thing: couldn’t make it work in 7.5, a 7.3 install worked fine, but upgrading to 7.4 broke it.

**Reply by MagnetarRocket** [reply]
I have been able to install & get up a working 7.5 virtual system using `.iso`; but it's CLI only, whole virtual system apparently hangs when i try `startx` with it.

**Reply by michealp-coder** [reply]
@macuserguru would you mind sharing in more detail what you did to get it to work (what were all of your UTM settings)

**Comment by a-r-o-n-d** [comment]
Curious if the UTM settings required to install OpenBSD 7.5 were ever resolved / fully laid out anywhere? I'm struggling to get through the install successfully. Thanks in advance!

**Reply by sysaulab** [reply]
I have some time today to give it a try. Can you provide details of your installation options and the problem you encounter?

**Reply by sysaulab** [reply]
When installing for m1:
- virtualize
- Others
- Everything as default…
- Edit the vm
- Change the boot media of the usb drive from “cd/dvd” to “Disk Image” instead of cdrom, otherwise it will not boot. ISO images dont seems to be available for arm64 at the moment.

So far so good, will reply with the notes regarding x11 when i get there.

**Reply by a-r-o-n-d** [reply]
That does seem to launch the installation program, thank you. With my previous settings the VM window would just remain black. I'll try getting through the OpenBSD install and hopefully have success.

**Reply by sysaulab** [reply]
X11 is only yielding black screens to me. Changing devices do not seems to help. Disabling the viogpu device as suggested on a forum (with ukc) did not solve the issue for me now. If you need gui and do not expect to play 3d games, xrdp will yield excellent performance and should be easy enough to get going.

let me know if you get stuck again or find a solution to the gpu issue so it can be added here.

**Reply by sysaulab** [reply]
I will attempt to create fix and submit a patch to solve this issue. If you have suggestions or comments about your experience installing openbsd on utm let me know.

**Reply by sysaulab** [reply]
Under x86-64, the usb3 keyboard is not working. I disabled usb from input to allow installation. X11 works without a flaw using virtio-ramfb. This suggest to me that the display problem is probably a bug specific to openbsd on arm64 and not UTM.

**Reply by a-r-o-n-d** [reply]
@sysaulab I really appreciate the help... I got through the OpenBSD 7.5 install and it's working well for my purposes. I haven't tried X11 and don't require a GUI since I'm only using this for a server inside a UTM machine.

**Reply by sysaulab** [reply]
In interest of helping newcomers experiencing problems with openbsd, I created a preinstalled OpenBSD VM repository at https://github.com/sysaulab/UTMBSD/releases/tag/openbsd

**Comment by davidhewett** [comment]
For anyone wondering about making X11 work on 7.5 or 7.4, disabling viogpu did work for me, by following these instructions:

First, make sure you have both the virtual display *and* serial console enabled.
Then, at the `boot>` prompt, type `boot sda0:/bsd -c` (this command may vary depending on what your disk names are).
At the resulting `UKC>` prompt in the serial console, type `disable viogpu`, and then `quit`.
Once it finishes booting (may take a second), type `startx` on the display console.

**Reply by MagnetarRocket** [reply]
Will it work with 7.6?

**Reply by gretel** [reply]
> Will it work with 7.6?

can't say for a new install but upgrading from 7.5 went fine for me.

**Reply by MagnetarRocket** [reply]
But i have a more earnest issue- how do i make sure that disabling `viogpu` sticks around for future boots?

**Comment by naveenrajm7** [comment]
Hello, 

I have built a default install <sup>1</sup> OpenBSD 7.6  with a simple [auto install conf file](https://github.com/naveenrajm7/utm-box/blob/main/packer_templates/http/openbsd/install.conf) using [UTM packer plugin](https://github.com/naveenrajm7/packer-plugin-utm).
You can use this base box in Vagrant with [UTM vagrant plugin](https://github.com/naveenrajm7/vagrant_utm). Check example [Vagrant file for OpenBSD](https://github.com/naveenrajm7/utm-box/blob/main/example_vagrantfiles/openbsd/Vagrantfile)

I have hosted the VM at [HCP Vagrant registry](https://portal.cloud.hashicorp.com/services/vagrant/registries/utm/boxes/openbsd-7.6), you can use it with Vagrant or download, untar the box file and use the .utm file directly with UTM.

You can also build your own version using the [Packer recipe for OpenBSD](https://github.com/naveenrajm7/utm-box/tree/main?tab=readme-ov-file#examples)

><sup>1</sup> - Only two remote holes in the default install, in a heck of a long time! - OpenBSD

**Comment by jbholland** [comment]
I came back to this. I used the instructions here (on a 7.7 arm install) 

First, make sure you have both the virtual display and serial console enabled.
Then, at the boot> prompt, type boot sda0:/bsd -c (this command may vary depending on what your disk names are).
At the resulting UKC> prompt in the serial console, type disable viogpu, and then quit.
Once it finishes booting (may take a second), type startx on the display console.


then I found a way to make it permanent here:

https://www.alextsang.net/articles/20200719-042436/index.html   (edit the commands, you want to disable viogpu). 

now I can start openbsd from UTM on my M1 MacBook and startx !

**Reply by jbholland** [reply]
this gives 800x600. I'm trying to find a way to get a higher resolution but no luck so far. advice welcome!

**Reply by jbholland** [reply]
it's also very slow. I can configure it to allow X over ssh and run a hi-resolution session that way but it's horribly slow. The QEMU 800x600 display is faster but still slow. really it's too slow to use. But things are kind of working! Makes me curious what it would be like running directly on arm.

**Reply by morgant** [reply]
Looking forward to giving this a try! ~I see the `-c` option in the OpenBSD bootloader is new as of [the {amd64,arm64}/7.7-release](https://www.openbsd.org/77.html).~

Any reason you're using `startx` instead of the more-secure `xenodm`? As for performance, I don't believe [OpenBSD/arm64](https://www.openbsd.org/arm64.html) has support for GPU acceleration -- please correct me if I'm mistaken! -- so it's probably limited to a virtual frame buffer. Should be fine for normal use, but any X11 application that normally takes advantage of GPU acceleration (even web browsers) will suffer.

I run my OpenBSD -current dev/porting/testing environment in VM hosted by OpenBSD's vmm (which doesn't even have frame buffer support, so a similar situation) and have xenodm(1), Xfvb(1), and x11vnc(1) configured to provide easy access to a virtual display:

https://marc.info/?l=openbsd-misc&m=168971104402240&w=2

The benefit of this particular implementation is that it should work with any resolution and VNC client, plus supports the xenodm(1) login window and logging in/out of users, transitioning pretty seamlessly between X sessions. I connect over an SSH tunnel to ensure I/O is encrypted.

**Reply by jholland-vin-dit** [reply]
Startx was just what I reached for to be able to test things. It's really too slow to use though :( . installing 7.7 on my old thinkpad tank today was really encouraging though.

**Reply by glenntam** [reply]
This worked for me,  thank you.

> then I found a way to make it permanent here:
> https://www.alextsang.net/articles/20200719-042436/index.html (edit the commands, you want to disable viogpu).

Instead of rc.shutdown, I was able to persist the change by adding the line "disable viogpu" to /etc/[bsd.re-config](https://man.openbsd.org/bsd.re-config) and [restarted twice](https://www.reddit.com/r/openbsd/comments/1jptdc6/anyone_using_bsd_reconfig/).

**Reply by huwr** [reply]
I tried this on a fresh 7.7, but after running `quit` at the UKC prompt, the display console is disabled and even after booting the console never comes back...

I also tried adding `disable viogpu` to `/etc/bsd.re-config` and restarting twice... with no joy.

**Comment by mordak** [comment]
Following up on this with an update.

I was able to do an entirely normal install of OpenBSD under UTM on a Macbook Air M2 without having to do anything special today. First step was install using the [install.iso](https://cdn.openbsd.org/pub/OpenBSD/snapshots/arm64/install78.iso) (**not** the install.img). It boots as a CD just fine into the installer, and you can install as normal.

- Create a new VM / Virtualize
- Choose Operating System / Other
- Choose CPU cores / RAM as you like (I did 4G ram and 2 CPU)
- Boot Device is CD/DVD, choose the install.iso
- Choose disk size as preferred (I did 64GB)
- Shared Directory I skipped
- Give it a name and Save

Launch the VM and it should boot into the installer. Do the install as usual, then Eject the CD/DVD, reboot, and it will boot into the OS. I did not change the default hardware, so it was default video device, network device, etc.

As of Jan 15, 2026 [OpenBSD now runs normally under Apple Hypervisor,](https://www.undeadly.org/cgi?action=article;sid=20260115203619) which means X just works. I was able to startx, install i3, etc., without any trouble. This is in [snapshots](https://cdn.openbsd.org/pub/OpenBSD/snapshots/arm64/) now, and should make it into the 7.9 release.

I did change one setting in the VM, and it was setting [display upscaling](https://docs.getutm.app/settings-qemu/devices/display/#upscalingdownscaling) to Linear instead of Nearest Neighbour, which made font anti-aliasing look good after fiddling some settings in .Xdefaults:

> Xft.antialias: 1
> Xft.hinting: 1
> Xft.hintstyle: hintslight
> Xft.rgpa: rgb

Others may get different results with different window managers, etc.

Overall the experience was very straightforward and works well. Machine is snappy and runs well as usual.
